composition-js/src/hints.ts

```diff
@@ -1,164 +1,186 @@
 import { SubgraphASTNode } from "@apollo/federation-internals";
 import { printLocation } from "graphql";
 
-export class HintID {
-  constructor(
-    readonly code: string,
-    readonly description: string,
-    readonly mainElementDescription: string
-  ) {
-  }
+export enum HintLevel {
+  WARN = 60,
+  INFO = 40,
+  DEBUG = 20,
+}
 
-  toString(): string {
-    return `${this.code}: ${this.description}`;
-  }
+export type HintCodeDefinition = {
+  code: string,
+  level: HintLevel,
+  description: string,
```

Comment:
We probably should consider that at some point in the future we may want to merge errors and hints into a single class as I think errors are really just hints that have a higher alerting level. Not sure if there's a case for keeping them separate.

Reply:
I generally agree, though there is a bunch of technical considerations to take into account when we do so (probably hard to unplug errors from `GraphQLError`, but also unclear using `GraphQLError` for debug level message is exactly "clean"; also, errors are use somewhat more pervasively than hints: maybe that gets in the way, maybe not, but worth looking). So I would prefer leaving any of that to future us for now.

As I mentioned, I do would want to try to at least get the doc merged, so "externally" it looks like those a different levels of the same thing, but I'm gonna call for this to be a followup too at this point.